pkg/component/autoscaling/clusterautoscaler/cluster_autoscaler_test.go

```diff
@@ -60,15 +60,41 @@ var _ = Describe("ClusterAutoscaler", func() {
 		image              = "registry.k8s.io/cluster-autoscaler:v1.2.3"
 		replicas     int32 = 1
 
-		machineDeployment1Name       = "pool1"
-		machineDeployment1Min  int32 = 2
-		machineDeployment1Max  int32 = 4
-		machineDeployment2Name       = "pool2"
-		machineDeployment2Min  int32 = 3
-		machineDeployment2Max  int32 = 5
-		machineDeployments           = []extensionsv1alpha1.MachineDeployment{
-			{Name: machineDeployment1Name, Minimum: machineDeployment1Min, Maximum: machineDeployment1Max},
-			{Name: machineDeployment2Name, Minimum: machineDeployment2Min, Maximum: machineDeployment2Max},
+		machineDeployment1Name           = "pool1"
+		machineDeployment1Min      int32 = 2
+		machineDeployment1Max      int32 = 4
+		machineDeployment1Priority       = ptr.To(int32(0))
+		machineDeployment2Name           = "pool2"
+		machineDeployment2Min      int32 = 3
+		machineDeployment2Max      int32 = 5
+		machineDeployment2Priority       = ptr.To(int32(40))
+		machineDeployment3Name           = "pool3"
+		machineDeployment3Min      int32 = 3
+		machineDeployment3Max      int32 = 5
+		machineDeployments               = []extensionsv1alpha1.MachineDeployment{
+			{Name: machineDeployment1Name, Minimum: machineDeployment1Min, Maximum: machineDeployment1Max, Priority: machineDeployment1Priority},
+			{Name: machineDeployment2Name, Minimum: machineDeployment2Min, Maximum: machineDeployment2Max, Priority: machineDeployment2Priority},
+			{Name: machineDeployment3Name, Minimum: machineDeployment3Min, Maximum: machineDeployment3Max},
+		}
+
+		workerConfig = []gardencorev1beta1.Worker{
```

Comment:
Can you also test the fallback case, i.e., priority defined in shoot worker but not yet set in `MachineDeployment`?
In addition, please consider a test case in which the fallback value cannot be found.